src/utils.ts

```diff
@@ -215,21 +219,64 @@ export const setCursor = (canvas: HTMLCanvasElement | null, cursor: string) => {
   }
 };
 
+let eraserCanvasCache: any;
+export const setEraserCursor = (
+  canvas: HTMLCanvasElement | null,
+  theme: AppState["theme"],
+) => {
+  const drawCanvas = () => {
+    const cursorImageSizePx = 20;
+    eraserCanvasCache = document.createElement("canvas");
+    eraserCanvasCache.height = cursorImageSizePx;
+    eraserCanvasCache.width = cursorImageSizePx;
+
+    const context = eraserCanvasCache.getContext("2d")!;
+    if (theme === THEME.DARK) {
+      context.strokeStyle = oc.white;
+    } else {
+      context.strokeStyle = oc.black;
+    }
+    context.beginPath();
+    context.arc(
+      eraserCanvasCache.width / 2,
+      eraserCanvasCache.height / 2,
+      5,
+      0,
+      2 * Math.PI,
+    );
+    context.stroke();
+  };
+  if (!eraserCanvasCache) {
+    drawCanvas();
+  } else {
+    const context = eraserCanvasCache.getContext("2d")!;
+    const strokeStyle = context.strokeStyle;
+    if (theme === THEME.LIGHT && strokeStyle !== oc.black) {
+      drawCanvas();
+    } else if (theme === THEME.DARK && strokeStyle !== oc.white) {
+      drawCanvas();
+    }
+  }
+
+  const previewDataURL = eraserCanvasCache.toDataURL(MIME_TYPES.svg) as DataURL;
+
+  setCursor(canvas, `url(${previewDataURL}) 4 4, auto`);
+};
+
 export const setCursorForShape = (
   canvas: HTMLCanvasElement | null,
-  shape: string,
+  appState: AppState,
 ) => {
   if (!canvas) {
     return;
   }
-  if (shape === "selection") {
-    resetCursor(canvas);
-  } else if (shape === "eraser") {
+  if (appState.elementType === "selection") {
     resetCursor(canvas);
-
+  } else if (appState.elementType === "eraser") {
+    setEraserCursor(canvas, appState.theme);
```

Comment:
we should call this conditionally, if the eraser cursor isn't already set — thought this may complicate the logic again (setting/resetting a local flag) — since the perf benefits of using a custom CSS cursor would be lost otherwise

Reply:
This is not needed since we are caching the canvas, data url